tests/compile/test_builders.py

```diff
@@ -584,6 +585,25 @@ def test_explicit_input_from_shared(self):
         out = test_ofg(y, y)
         assert out.eval() == 4
 
+    def test_repeated_inputs_raises(self):
+        x = pt.dscalar("x")
+        y = pt.dscalar("y")
+
+        with pytest.raises(
+            ValueError,
+            match="There following variables were provided more than once as inputs to the "
+            "OpFromGraph",
+        ):
+            OpFromGraph([x, x], [x + y])
+
+    def test_input_is_output_allowed_if_unused_inputs_ignored(self):
+        x = pt.dscalar("x")
+
+        g = OpFromGraph([x], [x], on_unused_input="ignore")
```

Comment:
I don't think this test is relevant? The problem were hitting was not because the output was just the input, was when we were creating an Op that took as inputs both the input and output of the original function, which happened to be the same in that case, so you had `OpFromGraph([x, x, ...], [...])`. 

I think `OpFromGraph([x], [x])` was never itself a problem durin this PR. I would use the test you have above instead, which should work when you pass the kwarg. You can even combine them if you want, although it's also fine separate?

Reply:
Sure, I can just combine them to a single test that has a pass and a fail case

Reply:
how does it look now